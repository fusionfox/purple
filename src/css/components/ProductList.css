.ProductList {
  margin: 10px;
}

.ProductList-title {
  font-weight: bold
}

.ProductList-list {
  list-style-type: none;
  padding: 0;
}
