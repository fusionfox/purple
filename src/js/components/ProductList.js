import React from 'react'
import '../../css/components/ProductList.css'
import SelectableItem from './SelectableItem'

const ProductList = (props) => {
  let products = props.products.map((product) => {
    return <li key={product.id} className='ProductList-product'>
      <SelectableItem
        name={product.name}
        id={product.id}
        handleItemSelection={props.handleProductSelection}
      />
    </li>
  })

  return (
    <figure className='ProductList'>
      <figcaption className='ProductList-title'>{props.title}</figcaption>
      <ul className='ProductList-list'>
        {products}
      </ul>
    </figure>
  )
}

export default ProductList
