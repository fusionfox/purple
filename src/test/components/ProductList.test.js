import React from 'react'
import renderer from 'react-test-renderer'
import ProductList from '../../js/components/ProductList'

const products = [
  {id: 'pr1', name: 'Number One'},
  {id: 'pr2', name: 'Number Two'}
]

describe('ProductList component', () => {
  it('Renders a list of products with a title', () => {
    const component = renderer.create(
      <ProductList
        title='HelloWorld'
        products={products}
        handleProductSelection={() => null}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
