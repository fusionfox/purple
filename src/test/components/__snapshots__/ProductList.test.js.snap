// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList component Renders a list of products with a title 1`] = `
<figure
  className="ProductList"
>
  <figcaption
    className="ProductList-title"
  >
    HelloWorld
  </figcaption>
  <ul
    className="ProductList-list"
  >
    <li
      className="ProductList-product"
    >
      <label
        className="SelectableItem-label"
      >
        <input
          className="SelectableItem-item"
          name="product"
          onChange={[Function]}
          type="checkbox"
          value="pr1"
        />
        Number One
      </label>
    </li>
    <li
      className="ProductList-product"
    >
      <label
        className="SelectableItem-label"
      >
        <input
          className="SelectableItem-item"
          name="product"
          onChange={[Function]}
          type="checkbox"
          value="pr2"
        />
        Number Two
      </label>
    </li>
  </ul>
</figure>
`;
